### PR TITLE
ENYO-2063: adapt to Node.js 0.10

### PR DESCRIPTION
**Pull request by asnowfix**
Tested on OSX. I do not expect any change on Linux or Windows.

**Demands** Node.js 0.10 (See https://enyojs.atlassian.net/browse/ENYO-2063 for details).
- ENYO-2063: clean-up needless `nodejs@0.8` code path (HEAD, origin/ENYO-2063, ENYO-2063)
- ENYO-2063: update npm dependencies & associated shrinkwrap
- ENYO-2063: use new patched node-unzip (0.1.7 + changes) for nodejs@0.10
- ENYO-2063: re-add node-unzip as submodule
- ENYO-2063: improve the overwrite, split fetch+unzip
- ENYO-2063: remove dependency on patched node-unzip

Enyo-DCO-1.1-Signed-off-by: Francois-Xavier KOWALSKI francois-xavier.kowalski@hp.com

### CODE REVIEW

File: ares-generator.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ var shell = require("shelljs"),
 
 	var generator = {};
 
+	var NODE_V_0_8 = !!process.version.match('^v0.8');
```
Comment (yves-del-medico) [inline ares-generator.js]: No longer needed and not used

File: ares-generator.js [hunk 2]
```diff
@@ -181,110 +190,100 @@ var shell = require("shelljs"),
...
 				log.info("generate#_processZipFile()", "Processing " + item.url);
-				
-				temp.mkdir({prefix: 'com.hp.ares.gen.processZipFile'}, (function(err, zipDir) {
+				var context = {
+					item: item
+				};
+				temp.mkdir({
+					prefix: 'com.enyojs.ares.generator',
+					suffix: ".d"
+				}, (function(err, tmpDir) {
```
Comment (yves-del-medico) [inline ares-generator.js:197]: err is not checked

File: ares-generator.js (hunk 1 above)
Comment (asnowfix) [inline ares-generator.js]: Cleaned-up one commit later

File: ares-generator.js (hunk 2 above)
Comment (asnowfix) [inline ares-generator.js:197]: fixed. commit uploaded.

File: ares-generator.js (hunk 1 above)
Comment (yves-del-medico) [inline ares-generator.js]: Still here

**Comment by asnowfix** [comment]
@yves-del-medico ready to merge

**Comment by yves-del-medico** [comment]
Merged. Tested on OSX: Chrome, Safari and Win7: Chrome, IE10.

**Pull request merged**